backend/lib/rpc-service/service.js

```diff
@@ -14,13 +14,20 @@ import { ClaimEvent, RawActorEvent, BlockEvent, RpcRespone } from './data-types.
  */
 export const rpcRequest = async (method, params) => {
   const reqBody = JSON.stringify({ method, params, id: 1, jsonrpc: '2.0' })
-  const response = await fetch(RPC_URL, {
-    method: 'POST',
-    headers: { 'content-type': 'application/json' },
-    body: reqBody
-  })
-
-  return Value.Parse(RpcRespone, (await response.json())).result
+  try {
+    const response = await fetch(RPC_URL, {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: reqBody
+    })
+    if (!response.ok) {
+      throw new Error(`Fetch failed - HTTP ${response.status}: ${await response.text()}`)
+    }
+    return Value.Parse(RpcRespone, (await response.json())).result
```

Comment:
```suggestion
    return Value.Parse(RpcRespone, await response.json()).result
```

Unnecessary wrapping